edgeless_node/src/lib.rs

```diff
@@ -359,6 +367,46 @@ async fn fill_resources(
                 });
             }
         }
+
+        if let (Some(host), Some(port), Some(messages_number_limit), Some(provider_id)) = (
```

Comment:
Would turn this into a struct that only requires one match.
Also might just put this into a separate config block.